tests/Integration/Momento.Sdk.Tests/TopicTest.cs

```diff
@@ -0,0 +1,236 @@
+        {
+            new byte[] { 0x01 },
+            new byte[] { 0x02 },
+            new byte[] { 0x03 },
+            new byte[] { 0x04 },
+            new byte[] { 0x05 }
+        };
+
+        var subscribeResponse = await topicClient.SubscribeAsync(cacheName, topicName);
+        Assert.True(subscribeResponse is TopicSubscribeResponse.Subscription,
+            $"Unexpected response: {subscribeResponse}");
+        var subscription = (TopicSubscribeResponse.Subscription)subscribeResponse;
+        
+        var testTask = Task.Run(async () =>
+        {
+            var messageCount = 0;
+            await foreach (var message in subscription)
+            {
+                Assert.NotNull(message);
+                Assert.True(message is TopicMessage.Item, $"Unexpected message: {message}");
+                if (((TopicMessage.Item)message).ValueByteArray!.SequenceEqual(skipMessage)) continue;
+                
+                Assert.Equal(valuesToSend[messageCount], ((TopicMessage.Item)message).ValueByteArray);
+
+                messageCount++;
+                if (messageCount == valuesToSend.Count)
+                {
+                    break;
+                }
+            }
+
+            subscription.Dispose();
+            return messageCount;
+        });
+        
+        // Send a few values to skip over to ensure the subscription received the messages under test
+        for (var i = 0; i < 5; i++)
+        {
+            await topicClient.PublishAsync(cacheName, topicName, skipMessage);
+            Thread.Sleep(100);
+        }
+        
+        foreach (var value in valuesToSend)
+        {
+            var publishResponse = await topicClient.PublishAsync(cacheName, topicName, value);
+            Assert.True(publishResponse is TopicPublishResponse.Success, $"Unexpected response: {publishResponse}");
+            Thread.Sleep(100);
+        }
+        foreach (var value in valuesToSend)
+        {
+            var publishResponse = await topicClient.PublishAsync(cacheName, topicName, value);
+            Assert.True(publishResponse is TopicPublishResponse.Success, $"Unexpected response: {publishResponse}");
+            Thread.Sleep(100);
+        }
+        
+        Assert.Equal(valuesToSend.Count, await testTask);
+    }
+
+    [Fact(Timeout = 5000)]
+    public async Task PublishAndSubscribe_String_Succeeds()
+    {
+        const string topicName = "topic_string";
+        var skipMessage = "skip";
+        var valuesToSend = new List<string>
+        {
+            "one",
+            "two",
+            "three",
+            "four",
+            "five"
+        };
+        
+        var subscribeResponse = await topicClient.SubscribeAsync(cacheName, topicName);
+        Assert.True(subscribeResponse is TopicSubscribeResponse.Subscription,
+            $"Unexpected response: {subscribeResponse}");
+        var subscription = (TopicSubscribeResponse.Subscription)subscribeResponse;
```

Comment:
Is this what we are expecting the end user code to look like? They will have to cast it?

I know that with the cache methods we were doing like `instanceof Subscription newVarName` and then it would smart cast. If there's not going to be a similar way for users to consume this then we may want to tweak how we are exposing the API, I don't want users to have to cast in the happy path.

Reply:
That's a holdover from my original testing. My documentation uses variable renaming but I wasn't thinking about it with the tests. I'll change it.